dffml/util/net.py

```diff
@@ -37,6 +40,19 @@ def __str__(self):
         return f"Protocol of URL {self.url!r} is not in allowlist: {self.allowlist!r}"
 
 
+class DirectoryNotExtractedError(Exception):
+    """
+    Raised when extraction of directory failed!
+    """
+
+    def __init__(self, directory_path):
+        super().__init__()
+        self.directory_path = directory_path
+
+    def __str__(self):
+        return f"Failed to extract - {self.directory_path}."
```

Comment:
- Exceptions usually don't end in periods
- With file paths it can be helpful to print with `repr()`, in case there are spaces or other odd characters the user may be having trouble noticing.

```suggestion
        return f"Failed to extract - {self.directory_path!r}"
```